mesonbuild/modules/qt.py

```diff
@@ -124,15 +124,41 @@ def __init__(self, interpreter: 'Interpreter', qt_version: int = 5):
         })
 
     def compilers_detect(self, state: 'ModuleState', qt_dep: 'QtDependencyType') -> None:
-        """Detect Qt (4 or 5) moc, uic, rcc in the specified bindir or in PATH"""
+        """Detect Qt (4, 5, or 6) moc, uic, rcc, lrelease binaries.
+
+        Look for them in these locations in this order:
+        * Locations specified in the config. Their names are formatted like this:
+          "qt<qt version>-<binary name>". For example, Qt5 moc is "qt5-moc".
```

Comment:
Using:
- `qt<qt version>-<binary name>`
- or as we already document it below, "tool" and "version", thus

  `qt<version>-<tool>`

is inconsistent with the existing searches we perform, which is in turn based on how Qt upstream distributes the tools and how distributions are known to have renamed it.

Expecting users to put the `qt<version>` part *first* when writing their cross file when it's never expected for the actual binary name to have it first, is not going to happen. So it would have to be reversed to go the other way around.